2 database related   500 database errors:  make 400 and ignore, respectively

these are 2 other 500 errors we see occasionally

Caused by: java.sql.SQLIntegrityConstraintViolationException: ORA-01400: cannot insert NULL into ("USERLOGINS5"."MULTIBLAST_JOBS"."QUERY")

 a way to reproduce this one is by entering a blast sequence like:
>def

**This should become a user error, a 400 with a nicer message**. 

--------------------
Caused by: java.sql.SQLIntegrityConstraintViolationException: ORA-00001: unique constraint (USERLOGINS5.MB_UQ_USER_JOB) violated

seems like a race condition
the constraint mb_uq_user_job  is in multiblast_users
see (https://github.com/VEuPathDB/service-multi-blast/blob/master/databases/wdk/user-tables.sql)
  CONSTRAINT mb_uq_user_job

the error means that it's trying to insert a link from a user to a job that already exists
ellie says:  so the error doesn't cause any problems, except for the 500

**unless i am missing something I would ignore this situation and not throw the 500.. since it is harmless...**

--- a/rest-service/src/main/kotlin/mb/lib/query/db/InsertUserLink.kt
+++ b/rest-service/src/main/kotlin/mb/lib/query/db/InsertUserLink.kt
@@ -9,25 +9,36 @@ data class InsertUserLink(private val con: Connection, private val row: UserBlas
   companion object {
     private const val Query = """
     INSERT INTO
-      userlogins5.multiblast_users (
-        job_digest
-      , user_id
-      , description
-      , max_download_size
-      , run_directly
+      userlogins5.multiblast_users
+    SELECT
+      ?, ?, ?, ?, ?
+    FROM
+      dual
+    WHERE
+      NOT EXISTS (
+        SELECT
+          1
+        FROM
+          userlogins5.multiblast_users
+        WHERE
+          job_digest = ?
+        AND
+          user_id = ?
       )
-    VALUES
-      (?, ?, ?, ?, ?)
     """
   }
 
   fun run() = BasicPreparedVoidQuery(Query, con, this::prep).execute()
 
   private fun prep(ps: PreparedStatement) {
-    ps.setBytes(1, row.jobID!!.bytes)
+    // SELECT
+    ps.setBytes(1, row.jobID.bytes)
     ps.setLong(2, row.userID)
     ps.setString(3, row.description)
     ps.setLong(4, row.maxDownloadSize)
     ps.setBoolean(5, row.runDirectly)
+    // WHERE NOT EXISTS
+    ps.setBytes(6, row.jobID.bytes)
+    ps.setLong(7, row.userID)
   }
 }

